main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidatorTest.java

```diff
@@ -17,110 +17,353 @@
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
 import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+import org.mobilitydata.gtfsvalidator.validator.TimepointTimeValidator.MissingTimepointColumnNotice;
+import org.mobilitydata.gtfsvalidator.validator.TimepointTimeValidator.MissingTimepointValueNotice;
 import org.mobilitydata.gtfsvalidator.validator.TimepointTimeValidator.StopTimeTimepointWithoutTimesNotice;
 
 public class TimepointTimeValidatorTest {
 
-  private static List<ValidationNotice> generateNotices(GtfsStopTime stopTime) {
+  private static GtfsStopTimeTableContainer createTable(
+      CsvHeader header, List<GtfsStopTime> stopTimes, NoticeContainer noticeContainer) {
+    return GtfsStopTimeTableContainer.forHeaderAndEntities(header, stopTimes, noticeContainer);
+  }
+
+  private static List<ValidationNotice> generateNotices(
+      CsvHeader header, List<GtfsStopTime> stopTimes) {
     NoticeContainer noticeContainer = new NoticeContainer();
-    TimepointTimeValidator validator = new TimepointTimeValidator();
-    validator.validate(stopTime, noticeContainer);
+    TimepointTimeValidator validator =
+        new TimepointTimeValidator(createTable(header, stopTimes, noticeContainer));
+    validator.validate(noticeContainer);
     return noticeContainer.getValidationNotices();
   }
 
+  private static CsvHeader createLegacyHeader() {
+    return new CsvHeader(
+        new String[] {
+          TRIP_ID_FIELD_NAME,
+          ARRIVAL_TIME_FIELD_NAME,
+          DEPARTURE_TIME_FIELD_NAME,
+          STOP_ID_FIELD_NAME,
+          STOP_SEQUENCE_FIELD_NAME,
+          STOP_HEADSIGN_FIELD_NAME,
+          PICKUP_TYPE_FIELD_NAME,
+          DROP_OFF_TYPE_FIELD_NAME,
+          CONTINUOUS_PICKUP_FIELD_NAME,
+          CONTINUOUS_DROP_OFF_FIELD_NAME,
+          SHAPE_DIST_TRAVELED_FIELD_NAME
+        });
+  }
+
+  private static CsvHeader createHeaderWithTimepointColumn() {
+    return new CsvHeader(
+        new String[] {
+          TRIP_ID_FIELD_NAME,
+          ARRIVAL_TIME_FIELD_NAME,
+          DEPARTURE_TIME_FIELD_NAME,
+          STOP_ID_FIELD_NAME,
+          STOP_SEQUENCE_FIELD_NAME,
+          STOP_HEADSIGN_FIELD_NAME,
+          PICKUP_TYPE_FIELD_NAME,
+          DROP_OFF_TYPE_FIELD_NAME,
+          CONTINUOUS_PICKUP_FIELD_NAME,
+          CONTINUOUS_DROP_OFF_FIELD_NAME,
+          SHAPE_DIST_TRAVELED_FIELD_NAME,
+          TIMEPOINT_FIELD_NAME
+        });
+  }
+
+  @Test
+  public void noTimepointColumn_noTimeProvided_shouldGenerateNotice() {
+    // unit test for legacy data that do not use timepoint column
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("first trip id")
+            .setArrivalTime(null)
+            .setDepartureTime(null)
+            .setStopId("stop id 0")
+            .setStopSequence(2)
+            .setTimepoint((Integer) null)
```

Comment:
Why do you need to explicitly set values to null? The same for other tests as well

Reply:
The goal is to test what happens when no value is provided for the field `timepoint` - the validator parses an empty csv cell as an `null` object when it comes to setting `stopTime.timepoint`, which is why `null` is used here. 

See, method `asEnum()` from `RowParser`: https://github.com/MobilityData/gtfs-validator/blob/60fc8561095734417f8d198ca73da41a3180de39/core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java#L304

Reply:
Thanks for the explanation! 

It looks like we may need to create a feature request to have an ability to specify test entities in the csv format, otherwise, we're trying to repeat logic of auto generated code, which may change and we end up with wrong tests.

Reply:
>It looks like we may need to create a feature request to have an ability to specify test entities in the csv format, otherwise, we're trying to repeat logic of auto generated code, which may change and we end up with wrong tests.

💯  I'd really like to see this. This would allow us to directly use real-world data that generates false-positives, etc. in the tests, reducing the likelihood of translation errors when trying to write code for the equivalent.

Reply:
Just to reinforce this - we really shouldn't be setting values to `null` in unit tests because that behavior will never actually happen in real-world execution of the project. As mentioned in https://github.com/MobilityData/gtfs-validator/pull/1044#discussion_r745858272, enums will always have a value, defaulting to `0`.

So the current code is fine for this PR, but we really should look at re-examining how these tests are set up.